staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -204,9 +204,24 @@ func (pf *PortForwarder) listenOnPortAndAddress(port *ForwardedPort, protocol st
 	return nil
 }
 
+// isUnwrappedIPv6Address indicates if we have an IPv6 that is not wrapped in
+// square brackets (needed, when using with port number). If host name, IPv4
+// address, or IPv6 address that is wrapped (which cannot be parsed as IP),
+// then return false.
+func isUnwrappedIPv6Address(nameOrIP string) bool {
```

Comment:
Whenever you find yourself writing a function like this, it should get your spidey senses tingling and you should spend a few minutes looking through the standard library.